bin/ex_libris_holdings_xml_parser.rb

```diff
@@ -0,0 +1,105 @@
+require 'nokogiri'
+
+class ExLibrisHoldingsXmlParser
+  def main(files)
+    files.each do |file|
+      File.open(file) do |io|
+        Nokogiri::XML(io).xpath("//record").each do |xml_record|
+          ht_record = HTRecord.new(xml_record)
+          # Skip records without ocns or where item_type is not print-y.
+          next if ht_record.oclc.empty?
+          next if ht_record.item_type == 'skip'
+          yield ht_record
+        end
+      end
+    end
+  end
+end
+
+class HTRecord
+  def initialize(xml_record)
+    # xml_record is a Nokogiri::XML::Element
+    @xml_record = xml_record
+  end
+
+  def self.header_tsv
+    %w[item_type oclc local_id status condition enum_chron issn govdoc].join("\t")
+  end
+
+  # Should use the same order as HTRecord.header_tsv
+  def to_tsv
+    [item_type, oclc, local_id, status, condition, enum_chron, issn, govdoc]
+      .join("\t")
+  end
+
+  # Given an xpath, return the text value of that xpath in @xml_record
+  def path_text(xpath)
+    @xml_record.xpath(xpath).text.strip
+  end
+
+  def oclc
+    @oclc ||= path_text("datafield[@tag='035']/subfield[@code='a']")
+      .scan(/\d+/).uniq.join(";")
+  end
+
+  def local_id
+    @local_id ||= path_text "controlfield[@tag='001']"
+  end
+
+  def condition
+    @condition ||= path_text "datafield[@tag='ITM']/subfield[@code='c']"
+  end
+
+  def item_type
+    @item_type ||= map_item_type(path_text "datafield[@tag='ITM']/subfield[@code='m']")
+  end
+
+  # Todo: Figure out the structure of enum_chrons.
+  def enum_chron
+    @enum_chron ||= [
+      path_text("datafield[@tag='ITM']/subfield[@code='a']"),
```

Comment:
with rubymarc we can do basically `record.fields("ITM").first["a"]` - should be easy enough just to use that.